nan.h

```diff
@@ -2157,6 +2157,32 @@ struct Tap {
 
 #undef TYPE_CHECK
 
+//=== Generic Maybefication ===================================================
+
+namespace imp {
+
+template <typename T> struct Maybefier;
+
+template <typename T> struct Maybefier<v8::Local<T> > {
+  static MaybeLocal<T> convert(v8::Local<T> v) {
+    return MaybeLocal<T>(v);
+  }
+};
+
+template <typename T> struct Maybefier<MaybeLocal<T> > {
+  static MaybeLocal<T> convert(MaybeLocal<T> v) {
+    return v;
+  }
+};
+
+}  // end of namespace imp
+
+template <typename T, template <typename> class CouldbeMaybe>
```

Comment:
Ok, it's official. There is no good free text on template template parameters.

However, _The Text_ on the subject is [C++ Templates - The Complete Guide](http://www.josuttis.com/tmplbook/), a book every C++ programmer should have.